### PR TITLE
Drop EOL RHEL, CentOS, Fedora, VirtuozzoLinux versions

### PR DESCRIPTION
**Pull request by jay7x**
* VirtuozzoLinux 7
* Fedora 38, 39
* Centos 7, 8
* RHEL 7

### CODE REVIEW

**Review by kenyon — APPROVED** [review]

**Comment by TheMeier** [comment]
Please update the PR title to list all affected OSes.

**Review by TheMeier — APPROVED** [review]

**Comment by jay7x** [comment]
Do we want to make a release before dropping EOL OSes?

File: metadata.json [hunk 1]
```diff
@@ -42,22 +42,13 @@
     {
       "operatingsystem": "RedHat",
       "operatingsystemrelease": [
-        "7",
```
Comment (ekohl) [inline metadata.json:45]: Technically RHEL 7 isn't really EOL. You can still buy ELS for another 4 years: https://www.redhat.com/en/blog/announcing-4-years-extended-life-cycle-support-els-red-hat-enterprise-linux-7

Whether we want to support that is a valid question.

**Review by ekohl — CHANGES_REQUESTED** [review]
If we're dropping this, I'd also like to drop the data files inhttps://github.com/voxpupuli/puppet-systemd/tree/master/data

File: metadata.json (hunk 1 above)
Comment (jay7x) [reply metadata.json:45]: We may offer paid support for RHEL7 in Voxpupuli modules then as well :-D

File: metadata.json [hunk 2]
```diff
@@ -87,13 +78,6 @@
     },
     {
       "operatingsystem": "Gentoo"
-    },
-    {
-      "operatingsystem": "Fedora",
-      "operatingsystemrelease": [
-        "37",
```
Comment (silug) [inline metadata.json:94]: Why drop Fedora support instead of updating the supported versions to `["40", "41"]`?

**Review by silug — CHANGES_REQUESTED** [review]
Why drop Fedora support instead of updating the supported versions to `["40", "41"]`?

**Comment by jay7x** [comment]
> Why drop Fedora support instead of updating the supported versions to `["40", "41"]`?

Hi @silug! I saw somewhere before that dropping EOL OSes and adding new OSes should be different PRs.. though.. maybe it was about different commits instead 🤔

File: data/Fedora.yaml [hunk 3]
```diff
@@ -1,8 +0,0 @@
----
-systemd::accounting:
```
Comment (bastelfreak) [inline data/Fedora.yaml:2]: my problem with this is that it actually breaks Fedora. Aren't the settings the same on all operating systems nowadays? can't we move that to init.pp instead?

Comment (jay7x) [reply data/Fedora.yaml:2]: I'm heading to redo the PR with adding new OSes before dropping old. So this file will stay.

**Comment by jay7x** [comment]
Converting to Draft. Will work on adding new OSes first (to keep Fedora files untouched e.g.).

**Comment by jay7x** [comment]
Should be ready to merge now

**Comment by bastelfreak** [comment]
Before we merge this, I would like to do a minor release to bring out all merged changes: https://github.com/voxpupuli/puppet-systemd/pull/501

**Comment by bastelfreak** [comment]
ah derp, we already have https://github.com/voxpupuli/puppet-systemd/pull/480 as a breaking change

**Review by bastelfreak — APPROVED** [review]

**Pull request merged**